### ISSUE TITLE
Multiplexer rendering multiple items into individual slots

### ISSUE THREAD

**Issue by sjmiles** [PolymerLabs/arcs#2140]
![image](https://user-images.githubusercontent.com/98045/47458424-59663b00-d78e-11e8-8977-ead61ccaeaa4.png)

I suspect it's because a second rendering task starts before the first one is completed.

**Comment by sjmiles** [comment]
Hrm, at least on the search page, doing more searching continues to add items to each sub-slot, so that suggests the old items are not being cleared (and not necessarily overlapping rendering tasks).

**Comment by piotrswigon** [comment]
I will take a stab at this this afternoon.

**Comment by piotrswigon** [comment]
We're having some issues with managing DOM containers in SlotComposer and friends.

Out of curiosity @sjmiles, did we change something in particle rendering that instead of replacing entire DOM trees, does minimum updates to the existing DOM tree (from the previous render)? This would explain what I see happening. Either way working on the fix.

**Comment by piotrswigon** [comment]
I traced the issue to #2099.

`SlotDomConsumer.createNewContainer(...)` now return a `shadowRoot` of the `div` container.

However `SlotDomConsumer.deleteContainer(...)` is still:

```
  deleteContainer(container) {
    if (container.parentNode) {
      container.parentNode.removeChild(container);
    }
  }
```

shadowRoot does not have a `parentNode`, so old containers are never cleaned up, which causes the behavior you see.

I don't know all that much about shadow DOM, so now that we know what's the problem maybe @sjmiles you can go over the methods in `SlotDomConsumer` operating on containers and figure out what should they look like in the Shell 0.6 world?

BTW an easy temporary workaround is to add `container.textContent = '';` in the `deleteContainer` method. But this obviously causes a buildup old obsolete DOM in the tree over time.

**Comment by piotrswigon** [comment]
@sjmiles Let me know whether you would like to take a stab at fixing this, or whether I should run with some proposal.

**Comment by sjmiles** [comment]
@piotrswigon thanks for the direction ... I'll take a stab at it.

**Comment by sjmiles** [comment]
Fixed by #2267
